### PR TITLE
[BC Break] Refactor and injector interface changes

### PR DESCRIPTION
**Pull request by michalbundyra**
During work on https://github.com/zendframework/zend-expressive-tooling/pull/12 discovered that usage injector of this library is quite hard, because inject/remove methods requires `IOInterface` from composer to output results there.

This PR refactor the library slightly, add QA tools and change interface of injector.

The main commit with change interface is 774cc6e.
Now `inject` and `remove` methods do not require `IOInterface` parameter and return bool on to indicate success or skipped operation.

`ComponentInstaller` interface is not changed, added there try...catch and output message based on response from methods `inject`/`remove`.

As interface of injector is changed this PR contains BC Break, so it should be released with version 0.7.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/test export-ignore
+.coveralls.yml export-ignore
+.gitattributes export-ignore
```
Comment (weierophinney) [inline .gitattributes:3]: Add `.docheader` to this list.

File: src/Injector/AbstractInjector.php [hunk 2]
```diff
@@ -188,29 +187,29 @@ public function inject($package, $type, IOInterface $io)
...
 
     /**
-     * Inject component $package after all dependencies into $config and return true.
-     * If any of dependencies is not registered the method will write error to $io
-     * and will also return true, to prevent injecting this package later.
-     * Method return false only in case when dependencies for the package are not found.
+     * Injects component $package after all dependencies
+     * into $config and returns true.
+     * If any of dependencies is not registered the method
+     * throws RuntimeException.
```
Comment (weierophinney) [inline src/Injector/AbstractInjector.php:198]: Why is this wrapped so short? (I can fix on merge, however...)

File: test/ConfigDiscoveryTest.php [hunk 3]
```diff
@@ -87,24 +95,24 @@ public function createModulesConfig()
     public function assertOptionsContainsNoopInjector(Collection $options)
     {
         if ($options->isEmpty()) {
-            throw new ExpectationFailedException('Options array is empty; no NoopInjector found!');
+            $this->fail('Options array is empty; no NoopInjector found!');
```
Comment (weierophinney) [inline test/ConfigDiscoveryTest.php:98]: When writing custom assertions, the PHPUnit documentation recommends throwing the above exception over using `fail()`. (Similarly, it recommends `assertThat()` vs the other assertion methods.)

I'll revert on merge.

**Review by weierophinney — APPROVED** [review]
I've noted a few changes, but most particularly the changes to the custom assertion methods. I can take care of all of these on merge, however.

File: test/ConfigDiscoveryTest.php (hunk 3 above)
Comment (michalbundyra) [reply test/ConfigDiscoveryTest.php:98]: Not sure if it is gonna work with PHPUnit 6.0:
https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/ExpectationFailedException.php#L21

Comment (weierophinney) [reply test/ConfigDiscoveryTest.php:98]: In my merge branch, I added a file `test/_autoload.php` that does a `class_alias` of the non-namespaced version to the namespaced variant, and added an `autoload-dev.files` entry for this file. Works like a charm!

**Pull request merged**